components/builder-originsrv/src/data_store.rs

```diff
@@ -634,6 +634,156 @@ impl DataStore {
         }
     }
 
+    pub fn create_origin_private_encryption_key(
+        &self,
+        opekc: &originsrv::OriginPrivateEncryptionKeyCreate,
+    ) -> SrvResult<originsrv::OriginPrivateEncryptionKey> {
+        let conn = self.pool.get(opekc)?;
+        let opek = opekc.get_private_encryption_key();
+        let rows = conn.query(
+            "SELECT * FROM insert_origin_private_encryption_key_v1($1, $2, $3, $4, $5, $6)",
+            &[
+                &(opek.get_origin_id() as i64),
+                &(opek.get_owner_id() as i64),
+                &opek.get_name(),
+                &opek.get_revision(),
+                &format!("{}-{}", opek.get_name(), opek.get_revision()),
+                &opek.get_body(),
+            ],
+        ).map_err(SrvError::OriginPrivateEncryptionKeyCreate)?;
+        let row = rows.iter().nth(0).expect(
```

Comment:
Check row len instead of expect?

Reply:
I think the option here is leave it as is or do something like:
```
if rows.len() != 0 {
    let row = rows.iter().nth(0).unwrap();
     Ok(self.row_to_origin_private_encryption_key(row))
} else {
    Err("some error")
}```

Does that sound right?